cmd/zfs/zfs_main.c

```diff
@@ -4383,10 +4383,66 @@ zfs_do_snapshot(int argc, char **argv)
 	return (-1);
 }
 
+typedef struct zfs_send_exclude_arg {
+	size_t count;
+	char **list;
+} zfs_send_exclude_arg_t;
+
+static void
+add_dataset_excludes(char *in_exclude, zfs_send_exclude_arg_t *context)
+{
+	char *exclude = strdup(in_exclude);
+	size_t tempSize = context->count;
+	char **tempArray = context->list;
+	char *tempString = NULL;
+
+	for (tempString = strsep(&exclude, ",");
+	    tempString != NULL;
+	    tempString = strsep(&exclude, ",")) {
+		tempArray = realloc(tempArray,
+		    (sizeof (char **)) * (tempSize + 1));
+		if (tempArray == NULL)
+			nomem();
+		tempArray[tempSize] = strdup(tempString);
+		if (tempArray[tempSize] == NULL)
+			nomem();
+		tempSize++;
+	}
+
+	free(exclude);
```

Comment:
Isn't `exclude` being progressively moved forward by the `strsep` calls?

> This token is terminated by overwriting the
       delimiter with a null byte ('\0'), and *stringp is updated to
       point past the token.

( https://man7.org/linux/man-pages/man3/strsep.3.html )

In that case, we might be leaking the allocation. Wondering why `free` doesn't complain...

Reply:
And my test program didn't show any leaks... I must have had a slightly different version.  Hm.  Anyway.